x/circuit/README.md

```diff
@@ -0,0 +1,126 @@
+# Circuit Breaker - block specific functionality in the state machine
+
+
+## Concepts
+
+Circuit Breaker is a module that is meant to avoid a chain needing to halt/shut down in the presence of a vulnerability, instead the module will allow specific messages or all messages to be disabled. When operating a chain, if it is app specific then a halt of the chain is less detrimental, but if there are applications built on top of the chain then halting is expensive due to the disturbance to applications. 
+
+Circuit Breaker works with the idea that an address or set of addresses have the right to block messages from being executed and/or included in the mempool. The address that disabled the message from being executed also has the right to reset the circuit breaker and allow the application to resume execution of the message. 
+
+## State
+
+### Accounts
+
+* AccountPermissions `0x1 | account_address  -> ProtocolBuffer(Access)`
+
+```go
+type level int32
+
+const (
+	// LEVEL_NONE_UNSPECIFIED indicates that the account will have no circuit
+// breaker permissions.
+    LEVEL_NONE_UNSPECIFIED = iota
+    // LEVEL_SOME_MSGS indicates that the account will have permission to
+    // trip or reset the circuit breaker for some Msg type URLs. If this level
+    // is chosen, a non-empty list of Msg type URLs must be provided in
+    // limit_type_urls.
+    LEVEL_SOME_MSGS
+    // LEVEL_ALL_MSGS indicates that the account can trip or reset the circuit
+    // breaker for Msg's of all type URLs.
+    LEVEL_ALL_MSGS 
+    // LEVEL_SUPER_ADMIN indicates that the account can take all circuit breaker
+    // actions and can grant permissions to other accounts.
+    LEVEL_SUPER_ADMIN
+)
+
+type Access struct {
+	level int32 
+	msgs []string // if full permission, msgs can be empty
+}
+```
+
+### Params
+
+* Params `0x2  -> Params`
+
+```go
+type params struct {
+	authority string // account which has permissions to disable/enable all messages and give permissions to other accounts
+}
+```
+
+### Disable List
+
+List of type urls that are disabled.
+
+* DisableList `0x3  -> []string` <!--- should this be stored in json to skip encoding and decoding each block, does it matter?-->
```

Comment:
```suggestion
* DisableList `0x3  msg_type_url -> []byte{}` <!--- should this be stored in json to skip encoding and decoding each block, does it matter?-->
```

We should probably just have an entry per message in the disable list. This can be quickly checked msg by msg rather then having to decode a whole blob.

Reply:
make sense.